app/controllers/course/assessment/submissions_controller.rb

```diff
@@ -27,14 +27,26 @@ def edit
   end
 
   def update
+    if params[:submit_answer]
```

Comment:
I don't think you can submit a single answer for a worksheet. It's all or nothing.